src/transformers/pipelines.py

```diff
@@ -404,9 +404,12 @@ def save_pretrained(self, save_directory):
         """
         Save the pipeline's model and tokenizer to the specified save_directory
         """
-        if not os.path.isdir(save_directory):
-            logger.error("Provided path ({}) should be a directory".format(save_directory))
+        if os.path.isfile(save_directory):
+            logger.error("Provided path ({}) should be a directory, not a file".format(save_directory))
             return
+        elif not os.path.exists(save_directory):
+            os.path.makedirs(save_directory)
+        assert os.path.isdir(save_directory), f"Failed to find or create {save_directory}"
```

Comment:
If `os.path.makedirs` fails, it will throw, no? So no need for the assert?

Reply:
I left it for readability but I think you are right, it doesn't help readability much.

Reply:
```
elif not os.path.exists(save_directory):
    os.path.makedirs(save_directory)
```
can be simplified to just 
```
os.path.makedirs(save_directory, exist_ok=True)
```
no?